modules/common/src/CacheableResponseTrait.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Drupal\common;
+
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Cachable Response Trait.
+ */
+trait CacheableResponseTrait {
+
+  /**
+   * Cache page max age config value.
+   *
+   * @var int
+   */
+  protected $cacheMaxAge;
+
+  /**
+   * Adds cache headers to the response.
+   *
+   * TODO: implement more flexible caching and move the code out of the trait.
+   *
+   * @param \Symfony\Component\HttpFoundation\Response $response
+   *   Symfony response.
+   *
+   * @return \Symfony\Component\HttpFoundation\Response
+   *   Symfony response.
+   *
+   * @throws \Exception
+   */
+  private function addCacheHeaders(Response $response) : Response {
+    $this->setCacheMaxAge();
+
+    if ($this->cacheMaxAge !== 0) {
+      $response->setCache([
```

Comment:
this section needs test coverage

Reply:
Done. Created tests to check the caching headers for JSON, YAML, CSV and HTML responses